arcgis-ios-sdk-samples/Layers/Browse OGC API feature service/BrowseOGCAPIServicesViewController.swift

```diff
@@ -0,0 +1,224 @@
+        
+        // The text field for OGC API service URL.
+        alertController.addTextField { [self] textField in
+            textField.placeholder = defaultServiceURL.absoluteString
+            textField.text = defaultServiceURL.absoluteString
+            textField.keyboardType = .URL
+            textFieldObserver = NotificationCenter.default.addObserver(
+                forName: UITextField.textDidChangeNotification,
+                object: textField,
+                queue: .main
+            ) { [unowned loadAction] _ in
+                let text = textField.text!.trimmingCharacters(in: .whitespacesAndNewlines)
+                // Enable the load button if textfield is not empty.
+                loadAction.isEnabled = !text.isEmpty
+            }
+        }
+        present(alertController, animated: true)
+    }
+    
+    @IBAction func browseLayerInfos(_ sender: UIBarButtonItem) {
+        let selectedIndex = featureCollectionInfos.firstIndex { $0 == selectedInfo }
+        let optionsViewController = OptionsTableViewController(labels: featureCollectionInfos.map { $0.title }, selectedIndex: selectedIndex) { [self] newIndex in
+            let selectedInfo = featureCollectionInfos[newIndex]
+            self.selectedInfo = selectedInfo
+            loadLayer(info: selectedInfo)
+        }
+        optionsViewController.modalPresentationStyle = .popover
+        optionsViewController.preferredContentSize = CGSize(width: 300, height: 300)
+        optionsViewController.presentationController?.delegate = self
+        optionsViewController.popoverPresentationController?.barButtonItem = sender
+        present(optionsViewController, animated: true)
+    }
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["BrowseOGCAPIFeatureServiceViewController"]
+        // Ask user for service URL when the sample has loaded.
+        askUserForServiceURL { [weak self] result in
+            switch result {
+            case .success(let serviceURL):
+                self?.loadService(url: serviceURL)
+            case .failure(let error):
+                self?.presentAlert(error: error)
+            }
+        }
+    }
+}
+
+// MARK: - UIAdaptivePresentationControllerDelegate
+
+extension BrowseOGCAPIFeatureServiceViewController: UIAdaptivePresentationControllerDelegate {
+    func adaptivePresentationStyle(for controller: UIPresentationController, traitCollection: UITraitCollection) -> UIModalPresentationStyle {
+        return .none
+    }
+}
+
+// MARK: - Custom errors
+
+extension BrowseOGCAPIFeatureServiceViewController {
+    private enum InputError: LocalizedError {
+        var errorDescription: String? {
+            switch self {
+            case .invalidURL:
+                return "Bad URL."
```

Comment:
```suggestion
                return "Invalid URL."
```

Reply:
I'll leave this to Phil.

- The SDK seems to give error with description "Bad URL" if the URL is 404
- I'm not sure about the best phrasing for a string that cannot be constructed into a URL 😅